packages/flutter/test/widgets/nested_scroll_view_test.dart

```diff
@@ -676,6 +678,233 @@ void main() {
+      child: Localizations(
+        locale: const Locale('en', 'US'),
+        delegates: const <LocalizationsDelegate<dynamic>>[
+          DefaultMaterialLocalizations.delegate,
+          DefaultWidgetsLocalizations.delegate,
+        ],
+        child: MediaQuery(
+          data: const MediaQueryData(),
+          child: NestedScrollView(
+            key: globalKey,
+            body: Container(),
+            headerSliverBuilder: (_, __) => <Widget>[const SliverAppBar()],
+          ),
+        ),
+      ),
+    ));
+
+    expect(globalKey.currentState.innerController, isNotNull);
+    expect(globalKey.currentState.outerController, isNotNull);
+  });
+
+  // One scrollExtent should be smaller than the height of the toolbar
+  // to ensure that the inner scroll controller is not scrolled in that case.
+  for (final double scrollExtent in const <double>[
+    kToolbarHeight - 9,
+    4.2e2,
+    1e3,
+    1e4
+  ]) {
+    testWidgets(
+        'NestedScrollViewState exposed scroll controllers work properly with a scroll extent of $scrollExtent',
+            (WidgetTester tester) async {
+          final GlobalKey<NestedScrollViewState> globalKey = GlobalKey();
+          final ScrollController scrollController = ScrollController();
+
+          await tester.pumpWidget(Directionality(
+            textDirection: TextDirection.ltr,
+            child: Localizations(
+              locale: const Locale('en', 'US'),
+              delegates: const <LocalizationsDelegate<dynamic>>[
+                DefaultMaterialLocalizations.delegate,
+                DefaultWidgetsLocalizations.delegate,
+              ],
+              child: MediaQuery(
+                data: const MediaQueryData(),
+                child: NestedScrollView(
+                  controller: scrollController,
+                  key: globalKey,
+                  body: ListView.builder(
+                    itemBuilder: (BuildContext context, int index) {
+                      return Text('$index');
+                    },
+                  ),
+                  headerSliverBuilder: (_, __) => <Widget>[const SliverAppBar()],
+                ),
+              ),
+            ),
+          ));
+
+          // The scroll gesture should be taken where in the inner body, so the whole scroll view is scrolled.
+          final TestGesture gesture = await tester.startGesture(const Offset(0, kToolbarHeight + 1));
+          await gesture.moveBy(Offset(0, -scrollExtent));
+
+          await tester.pump();
+
+          expect(
+            globalKey.currentState.innerController.position.pixels +
+                globalKey.currentState.outerController.position.pixels,
+            scrollExtent,
+          );
+
+          // The outer scroll view cannot scroll past its height, which should be kToolbarHeight as a SliverAppBar represents the outer body.
+          expect(
+            globalKey.currentState.outerController.position.pixels,
+            min(kToolbarHeight, scrollExtent),
```

Comment:
The test should verify that for a provided scrollExtent the position is at the right place - not that it is between a max and a min.

(same below).

Reply:
It just makes the tests more concise. I will need to add more loc to address this - it is just a condition I explained in the comment.